sdk/metric/controller/push/push.go

```diff
@@ -73,24 +75,26 @@ var _ Ticker = realTicker{}
 // using the provided integrator, exporter, collection period, and SDK
 // configuration options to configure an SDK with periodic collection.
 // The integrator itself is configured with the aggregation selector policy.
-func New(integrator export.Integrator, exporter export.Exporter, period time.Duration, opts ...Option) *Controller {
+func New(integrator export.Integrator, exporter export.Exporter, configNotifier *notifier.ConfigNotifier,
+	opts ...Option) *Controller {
```

Comment:
Instead of introducing a new field for the configNotifier, what do you think about bundling it in with the Options logic instead? That way, maybe we can maintain compatibility with older code and don't have to worry about breaking existing projects that use `push.New()`?

Reply:
Thanks for the look over!

Your first point got me thinking whether ConfigNotifier should be a mandatory field in the Controller. I talked it over with @vmingchen, and we decided that it should be mandatory, with the option for a config that is constant that we never change. This means we can't fold it into the Options logic, but in any case, thanks for bringing this up! It brought up an important point.

As to your second point, I think this might be problematic in that we'd have to register the notifier with every single different exporter that ends up getting implemented for open telemetry, and we want to try to keep as much as the logic inside the controller and dynamicconfig modules as we can.